pkg/highlight/highlighter.go

```diff
@@ -350,10 +355,11 @@ func (h *Highlighter) ReHighlightStates(input LineStates, startline int) int {
 
 	h.lastRegion = nil
 	if startline > 0 {
-		h.lastRegion = input.State(startline - 1)
+		h.lastRegion = input.StateNoLock(startline - 1)
```

Comment:
No locking at all here?
Looks like one more argument for restoring per-line locks for `state` and `match`.